datalad_gooey/gooey.py

```diff
@@ -49,26 +62,33 @@ class Gooey(Interface):
     @eval_results
     # signature must match parameter list above
     # additional generic arguments are added by decorators
-    def __call__(language='en'):
-        if language == 'en':
-            msg = 'Hello!'
-        elif language == 'de':
-            msg = 'Tachchen!'
+    def __call__(path: str = None):
+        if not path:
+            path = Path.cwd()
         else:
-            msg = ("unknown language: '%s'", language)
+            path = Path(path).resolve()
+        
+        qtapp = QApplication(sys.argv)
+        gooey = GooeyApp(path)
+        gooey.main_window.show()
+        # let a command run to have content appear in the console log
+        # uncomment for demo
+        #thread = MyThread().start()
+        sys.exit(qtapp.exec())
```

Comment:
```suggestion
        qtapp.exec()
```

If we do not make the python session exit after the app closes, we get to the result generation.

Reply:
But even with the change above, we have to do more de-initialization after the app closes. We also need to set a different DataLad UI backend again, otherwise result rendering would still target the GUI widget that is no longer shown.

Likely we should alter the app instance to remember the modifications it did, and undo them on exit. We can have a dedicated `deinit()` method that we call in the wrapper code for that.

Reply:
Ok, I'll have to sharpen up my understanding of pyside6/Qt for this first.

Reply:
To be clear, if it takes too long for me to figure this out I have no problem whatsoever if someone else wants to give it a go.

Reply:
OK, will do